dashboard/views.py

```diff
@@ -134,6 +137,90 @@ def get_course_info(request, course_id=0):
     return HttpResponse(json.dumps(resp, default=str))
 
 
+@permission_required('dashboard.update_course_info',
+                     fn=objectgetter(Course, 'course_id', 'canvas_id'), raise_exception=True)
+def update_course_info(request, course_id=0):
+    """
+
+    :param request: HTTP `PUT` req.; body should contain the JSON body…
+    :param course_id: Integer Canvas course ID number, typically six digits or less.
+    :return: HttpResponse containing `{"default": "success"}` or `{"default": "fail"}`
+    """
+    logger.info(update_course_info.__name__)
+
+    if (request.method != 'PUT'):
+        return HttpResponseBadRequest(JsonResponse({'error': 'Invalid request method.'}))
```

Comment:
I noticed that using `JsonResponse` like this is better than the wordy `HttpResponse(json.dumps({key: value}),content_type='application/json')` technique used in other parts of this code.

The drawback to `JsonResponse` is that it always has a HTTP response code of `200`.  However, I found that by wrapping it in another kind of response, like `HttpResponseBadRequest`, it's easy to change that in a way that's easy to understand what it's doing.

Also, I think more of the API entry points should verify they were requested with the correct HTTP method.  I **_thought_** there was a way to include the method required as part of the path given in `urls.py`, but that hasn't been used in MyLA before.  Either I've misremembered or it was decided that it's not a good technique for MyLA.